src/Autosuggest.js

```diff
@@ -235,15 +252,19 @@ class Autosuggest extends Component {
           case 'Enter': {
             const focusedSuggestion = this.getFocusedSuggestion();
 
-            if (focusedSuggestion !== null) {
+            if (focusedSuggestion) {
```

Comment:
Sure can. 

This is something new that has been introduced and wasn't possible before (but as I'll explain, is unavoidable as far as I can tell).

 Theoretically, it happens when a selection is hovered but that index is not in the suggestions. Of course, this can't happen on hover events because the suggestions are not visible (and don't exist). But `selectFirstSuggestion` acts as a hover event and selects/hovers the first suggestion `onChange`. Since `onChange` has no way to know if there are suggestions (suggestions don't exist until the next render cycle) it always sets the `focusedSuggestionIndex` to 0. This is the part that I believe is unavoidable.

So calling `this.getFocusedSuggestion()` will return `undefined` at the focused index, every time there are no suggestions given and enter is pressed:

![](http://dp.hanlon.io/1c0q190d3O3A/Screen%20Recording%202016-05-09%20at%2012.23%20PM.gif)

Of course a better solution here is to not set a `focusedSuggestionIndex` if the `suggestions` array is empty. I looked at this pretty extensively, and I found no way to determine this at the time `onChange` is called (since the suggestions are not fetched until the next render cycle).

Thus I think a fair strategy is to say: when enter is pressed and the focused selection doesn't exist in the selections array, do nothing.

Does that make sense?